codec/src/test/java/io/netty/handler/codec/compression/ZstdDecoderTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.compression;
+
+import com.github.luben.zstd.ZstdOutputStream;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import org.junit.Test;
+import org.junit.experimental.theories.DataPoints;
+
+import java.io.ByteArrayOutputStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class ZstdDecoderTest extends AbstractDecoderTest {
+
+    public ZstdDecoderTest() throws Exception {
+    }
+
+    @Override
+    public void initChannel() {
+        channel = new EmbeddedChannel(new ZstdDecoder());
+    }
+
+    @DataPoints("smallData")
+    public static ByteBuf[] smallData() {
+        ByteBuf direct = Unpooled.directBuffer(compressedBytesSmall.length);
+        direct.writeBytes(compressedBytesSmall);
+        return new ByteBuf[] {direct};
+    }
+
+    @DataPoints("largeData")
+    public static ByteBuf[] largeData() {
+        ByteBuf direct = Unpooled.directBuffer(compressedBytesLarge.length);
+        direct.writeBytes(compressedBytesLarge);
+        return new ByteBuf[] {direct};
```

Comment:
The codec should work with both types of memory: direct and heap. Zstd library supports both.
We also discussed that intergration tests should cover heap memory: https://github.com/netty/netty/pull/10422/files#r470280205